src/Symfony/Component/Security/Core/Authorization/Voter/Voter.php

```diff
@@ -19,11 +19,22 @@
  * @author Roman Marintšenko <inoryy@gmail.com>
  * @author Grégoire Pineau <lyrixx@lyrixx.info>
  *
- * @template TAttribute of string
+ * @template TAttribute of mixed
  * @template TSubject of mixed
  */
 abstract class Voter implements VoterInterface, CacheableVoterInterface
 {
+    public function __construct()
+    {
+        if ('string' === (string) (new \ReflectionMethod(static::class, 'supports'))->getParameters()[0]->getType()) {
```

Comment:
nice, remains the caching of the check in a static property, so that we run this once per class

Reply:
is it worth it? I mean rarely anyone will instantiate twice same service?

Reply:
Indeed, I forgot this was a service. Works for me then.